src/policy/rbf.cpp

```diff
@@ -180,3 +180,43 @@ std::optional<std::string> PaysForRBF(CAmount original_fees,
     }
     return std::nullopt;
 }
+
+std::optional<std::string> CheckMinerScores(CAmount replacement_fees,
+                                            int64_t replacement_vsize,
+                                            const CTxMemPool::setEntries& ancestors,
+                                            const CTxMemPool::setEntries& direct_conflicts,
+                                            const CTxMemPool::setEntries& original_transactions)
+{
+        const CFeeRate replacement_individual_feerate(replacement_fees, replacement_vsize);
+        // Ancestor feerate is the total modified fees divided by the total size. To get the
+        // ancestor feerate, add up all the individual modified fees and sizes. Don't try to use the
+        // cached ancestor fees and sizes because entries may have overlapping ancestors.
+        for (CTxMemPool::txiter it : ancestors) {
+            replacement_fees += it->GetModifiedFee();
+            replacement_vsize += it->GetTxSize();
+        }
+        const CFeeRate replacement_ancestor_feerate(replacement_fees, replacement_vsize);
+        // A package/transaction's ancestor feerate is not equivalent to the miner score; it may
+        // overestimate. Some subset of the ancestors could be included by itself if it has other
+        // high-feerate descendants or are themselves higher feerate than this package/transaction.
+        // For now, as a conservative estimate, use the minimum between the transaction's individual
+        // feerate and ancestor feerate.
+        const CFeeRate replacement_miner_score = std::min(replacement_individual_feerate, replacement_ancestor_feerate);
+        for (const auto& entry : direct_conflicts) {
+            const CFeeRate original_individual_feerate(entry->GetModifiedFee(), entry->GetTxSize());
```

Comment:
Note to self from Greg

For v3, where we can ask the exact miner score, use that instead of individual, otherwise absurd brutal fee when replacing child + switching sponsees. https://github.com/bitcoin/bitcoin/pull/26403/files#diff-d18bbdec91d0f4825b512a31f34666b000c7b7e2e05a3d43570c4b971532616fR415